grpc.go

```diff
@@ -504,9 +504,17 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 
 	select {
 	case exitCode := <-proc.exitCodeCh:
-		ctr.deleteProcess(proc)
-		proc.process.Wait()
-		proc.closePostExitFDs()
+		// proc.process.Wait() & proc.closePostExitFDs()
+		// should be called only once.
+		if ctr.deleteProcess(proc) {
+			proc.process.Wait()
+			proc.closePostExitFDs()
+		}
+		// There might be multiple WaitProcess() requested before
+		// the process exits. All of them should be able to get the
+		// exitCode. So send the exitCode back to proc.exitCodeCh
+		// for them.
```

Comment:
I am not sure why we want this change here. Could you elaborate ? I mean why should we allow several WaitProcess() on the same process, where is the real use case behind that ?
Also, I think this will simply not work, since your channel has been closed here: https://github.com/kata-containers/agent/blob/master/reaper.go#L114